core-bundle/src/InsertTag/Resolver/LinkInsertTag.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\InsertTag\Resolver;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsInsertTag;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\InsertTag\Exception\InvalidInsertTagException;
+use Contao\CoreBundle\InsertTag\InsertTagResult;
+use Contao\CoreBundle\InsertTag\OutputType;
+use Contao\CoreBundle\InsertTag\ResolvedInsertTag;
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
+use Contao\FrontendUser;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Symfony\Component\Routing\Exception\ExceptionInterface;
+
+class LinkInsertTag
+{
+    public function __construct(
+        private readonly ContaoFramework $framework,
+        private readonly TokenChecker $tokenChecker,
+    ) {
+    }
+
+    #[AsInsertTag('link')]
+    #[AsInsertTag('link_open')]
+    #[AsInsertTag('link_title')]
+    #[AsInsertTag('link_name')]
+    #[AsInsertTag('link_url')]
+    public function replaceInsertTag(ResolvedInsertTag $insertTag): InsertTagResult
+    {
+        $strTarget = null;
+        $strClass = '';
+
+        if (null === $insertTag->getParameters()->get(0)) {
+            throw new InvalidInsertTagException('Missing parameters for link insert tag.');
+        }
+
+        $urlParam = $insertTag->getParameters()->get(0);
+
+        // External links
+        if (str_starts_with($urlParam, 'http://') || str_starts_with($urlParam, 'https://')) {
```

Comment:
```suggestion
        if (\is_string($urlParam) && (str_starts_with($urlParam, 'http://') || str_starts_with($urlParam, 'https://'))) {
```

This will otherwise fail with a type error when you use `{{link_url::42}}`.

Reply:
I think I should remove the “magic” that converts the parameter in `{{link_url::42}}` to an integer, it only caused type problems so far without any use.